File: nff/commands/clean.py
"""nff clean — remove build artifacts."""

import shutil
import tempfile
from pathlib import Path

import click


@click.command()
def clean():
    """Remove nff build artifacts."""
    # nff_sketch = arduino-cli backend; nff_pio = PlatformIO backend (each scaffold's
    # heavy .pio/build output lives nested inside nff_pio, so one rmtree clears it).
    tmp = Path(tempfile.gettempdir())
    removed = []
    for name in ("nff_sketch", "nff_pio"):
        d = tmp / name
        if d.exists():
            shutil.rmtree(d, ignore_errors=True)
            removed.append(str(d))
    if removed:
        click.echo("\n".join(f"Removed {p}" for p in removed))
    else:
        click.echo("Nothing to clean.")

File: nff-rs/nff/src/commands/clean.rs
use anyhow::Result;

/// `nff clean` — remove nff build artifacts. `nff_sketch` is the arduino-cli backend's
/// temp dir; `nff_pio` is the PlatformIO backend's (each scaffold's heavy `.pio/build`
/// output lives nested inside, so one removal clears it). BYO pio projects keep their
/// own `.pio` — not nff's to delete.
pub fn run() -> Result<()> {
    let tmp = std::env::temp_dir();
    let mut removed = Vec::new();
    for name in ["nff_sketch", "nff_pio"] {
        let dir = tmp.join(name);
        if dir.exists() {
            std::fs::remove_dir_all(&dir).ok();
            removed.push(dir.display().to_string());
        }
    }
    if removed.is_empty() {
        println!("Nothing to clean.");
    } else {
        for p in removed {
            println!("Removed {p}");
        }
    }
    Ok(())
}
